c-svc/transform.c

```diff
@@ -0,0 +1,42 @@
+#include "transform.h"
+
+/* Adds the x, y and z from newVector to the observation array in the tso,
+ * If the tso.observation array is full we do nothing
+ * Note: tso is both input and output param 
+ */
+void addVectorToTimeSeriesObservation(TimeSeriesObservation *tso, double newVector[VECTOR_DIM]) {
+    int i = tso->_currentIndex;
+    
+    if(!tso->isFull) {
+        tso->observation[i  ] = newVector[0];
```

Comment:
```suggestion
        tso->observation[i]   = newVector[0];
```
Maybe we can make a loop 
```
for (j = 0; j < 3; j++){
   tso->observation[i+j] = newVector[j]
}
```

Reply:
I think it is less clear what is happening. What do you think?
```
        for(j = 0; j < VECTOR_DIM; j++) {
            tso->observation[i+j] = newVector[j];
        }
```